lib/server/services/run/createRun.js

```diff
@@ -30,7 +32,13 @@ exports.createRun = async (newRun) => {
             throw new ConflictError(`A run already exists with ${runNumber ? 'run number' : 'id'} ${runNumber ?? runId}`);
         }
     }
-
+    if (detectors) {
+        await RunDetectorsRepository.insertMany(detectors.map((detector) => ({
+            runNumber,
+            detectorId: detector.id,
+            quality: 'good',
+        })));
+    }
```

Comment:
It is at useCase level

Reply:
We negate every error thrown already in the try, catch.

Reply:
It is not that easy, because you do not know if this function (low level service) is run in a transaction or not. Therefore you should map this in a transaction, and take as parameter an optionally already existing transaction.
Have a look at `createLog`, and to see what happen you can change `updateRun` to remove the transaction transmission:
- Comment out transaction argument l. 83
- Trow an error after l.134

Then, change the quality of a good run to bad and see what happens